app/scripts/modules/ecs/src/serverGroup/configure/wizard/serviceDiscovery/ServiceDiscovery.tsx

```diff
@@ -0,0 +1,194 @@
+import React from 'react';
+import { module, IPromise } from 'angular';
+import { react2angular } from 'react2angular';
+import { Alert } from 'react-bootstrap';
+import { Option } from 'react-select';
+import {
+  IEcsServerGroupCommand,
+  IEcsServiceDiscoveryRegistry,
+  IEcsServiceDiscoveryRegistryAssociation,
+} from '../../serverGroupConfiguration.service';
+import { HelpField, TetheredSelect } from '@spinnaker/core';
+
+export interface IServiceDiscoveryProps {
+  command: IEcsServerGroupCommand;
+  notifyAngular: (key: string, value: any) => void;
+  configureCommand: (query: string) => IPromise<void>;
+}
+
+interface IServiceDiscoveryState {
+  serviceDiscoveryAssociations: IEcsServiceDiscoveryRegistryAssociation[];
+  serviceDiscoveryRegistriesAvailable: IEcsServiceDiscoveryRegistry[];
+}
+
+export class ServiceDiscovery extends React.Component<IServiceDiscoveryProps, IServiceDiscoveryState> {
+  constructor(props: IServiceDiscoveryProps) {
+    super(props);
+    const cmd = this.props.command;
+
+    this.state = {
+      serviceDiscoveryAssociations: cmd.serviceDiscoveryAssociations,
+      serviceDiscoveryRegistriesAvailable:
+        cmd.backingData && cmd.backingData.filtered ? cmd.backingData.filtered.serviceDiscoveryRegistries : [],
+    };
+  }
+
+  public componentDidMount() {
+    this.props.configureCommand('1').then(() => {
+      this.setState({
+        serviceDiscoveryRegistriesAvailable: this.props.command.backingData.filtered.serviceDiscoveryRegistries,
+      });
+    });
+  }
+
+  private getNameToRegistryMap = (): Map<string, IEcsServiceDiscoveryRegistry> => {
+    const displayNameToRegistry = new Map<string, IEcsServiceDiscoveryRegistry>();
+    this.props.command.backingData.filtered.serviceDiscoveryRegistries.forEach(e => {
+      displayNameToRegistry.set(e.displayName, e);
+    });
+
+    return displayNameToRegistry;
+  };
+
+  private getEmptyRegistry = (): IEcsServiceDiscoveryRegistry => {
+    return {
+      account: '',
+      region: '',
+      name: '',
+      id: '',
+      arn: '',
+      displayName: '',
+    };
+  };
+
+  private updateServiceDiscoveryRegistry = (index: number, newRegistry: Option<string>) => {
+    const registryMap = this.getNameToRegistryMap();
+    let newServiceDiscoveryRegistry = registryMap.get(newRegistry.value);
+
+    if (!newServiceDiscoveryRegistry) {
```

Comment:
since this is firing when the registry drop-down is changed, when would we not find the registry? could `newRegistry` be search input (vs. an existing option)?

Reply:
I was trying to maintain consistency with how the other properties are loaded in the configuration wizard currently (artifacts, container mappings, etc)